apps/src/netsim/NetSimRouterNode.js

```diff
@@ -1463,17 +1530,16 @@ NetSimRouterNode.prototype.forwardMessageToRecipient_ = function (message, onCom
     return;
   }
 
-  if (toAddress === this.getAddress()) {
-    // This packet has reached its destination, it's done.
-    logger.warn("Packet stopped at router.");
-    this.log(message.payload, NetSimLogEntry.LogStatus.SUCCESS);
+  var destinationNode = this.getNextNodeTowardAddress_(toAddress);
+  if (destinationNode === undefined) {
+    // Can't find or reach the address within the simulation
+    logger.warn("Destination address not reachable");
+    this.log(message.payload, NetSimLogEntry.LogStatus.DROPPED);
     onComplete(null);
     return;
-  }
-
-  var destinationNodeID = this.getNodeIDForAddress_(toAddress);
-  if (destinationNodeID === undefined) {
-    logger.warn("Destination address not in local network");
+  } else if (destinationNode === this && toAddress === this.getAddress()) {
+    // This router IS the packet's destination, it's done.
```

Comment:
The `toAddress === this.getAddress()` check works around the way the Auto-DNS runs on the router node but has its own address.